### PR TITLE
Fix and Enhanced Custom Variable

### PR DESCRIPTION
**Pull request by elfilibusterismo**

### CODE REVIEW

**Review by hasrat-ali — CHANGES_REQUESTED** [review]
Why did you add `ArrayList` to the variable category instead of the list category? And who uses List in the variable tab? Why didn't you include this in the list tab, while there is also a `custom` list?

**Comment by elfilibusterismo** [comment]
> Why did you add `ArrayList` to the variable category instead of the list category? And who uses List in the variable tab? Why didn't you include this in the list tab, while there is also a `custom` list?

Is there a problem with it? What if they want more customizable for list , they can still use it.   I just make the Custom Variable more flexible.

**Comment by hasrat-ali** [comment]
Screenshot_2023-07-13-10-15-14-71
Isn't this `more` customisable?

**Comment by elfilibusterismo** [comment]
> Screenshot_2023-07-13-10-15-14-71 Isn't this `more` customisable?

It is , but can u set the initializer tho.

I don't see any problem with it.

**Comment by elfilibusterismo** [comment]
Other than that, is there any problem or issue?

**Review by JavkhlanK — CHANGES_REQUESTED** [review]
and another thing to note for all files is to format the files just like the rest of the repo:
 - an indent has 4 spaces
 - after `if`s, `switch`es, etc. and before e.g. `{`s there should be a space
 - and also very important: no stray lines before a class' fields (like in CustomVariableUtil.java) as well as before one's last `}`

or to sum up: please (mostly) follow [Google Java Style](https://google.github.io/styleguide/javaguide.html).

File: app/src/main/java/mod/hasrat/menu/ExtraMenuBean.java [hunk 1]
```diff
@@ -684,6 +690,14 @@ private void defaultMenus(Ss menu) {
...
                     }
                 }
+                for(String variable : projectDataManager.e(javaName, 6)){
+                    String variableType = CustomVariableUtil.getVariableType(variable);
+                    String variableName = CustomVariableUtil.getVariableName(variable);
+                    if(menuName.equals(variableType)){
+                        title = "Select a " + variableType + " Variable";
+                        menus.add(variableName);
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (RO-ENPatcher) [reply app/src/main/java/mod/hasrat/menu/ExtraMenuBean.java:700]: It's for the Custom Variable(6) with modifiers, Old CustomVariable(5) didn't remove for backwards compatibility. It is just for new Custom Variable menu.

[the remark this answers is not in the document]
Comment (elfilibusterismo) [reply app/src/main/java/mod/hasrat/menu/ExtraMenuBean.java:700]: It's for the Custom Variable(6) with modifiers, Old CustomVariable(5) didn't remove for backwards compatibility. It is just for new Custom Variable menu.

**Pull request closed without merging**